### PR TITLE
Update MRS notebook to Build 11.0

### PR DESCRIPTION
**Pull request by drlaw1558**
This PR updates the MRS notebook to Build 11.0 (jwst version 1.15.1).  A few cells have changed accordingly; ramp fitting no longer requires parallelization, and pixel replacement moves to spec3.  Turned off a few parameters that are now set by default in the latest CRDS reference files, and turned off cosmic ray showers correction as a default in the notebook since it can fail for some programs.

### The First Cell

- [ ] The title of the notebook in a first-level heading (eg. `<h1>` or `# in markdown`).
- [ ] A brief description of the notebook.
- [ ] A table of contents in an [ordered list](https://www.markdownguide.org/basic-syntax/#ordered-lists) (`1., 2.,` etc. in Markdown).
- [ ] The author(s) and affiliation(s) (if relevant).
- [ ] The date first published.
- [ ] The date last edited (if relevant).
- [ ] A link to the notebook's source(s) (if relevant).

### The Rest of the Cells

- [ ] There is only one H1 (`#` in Markdown) used in the notebook.
- [ ] The notebook uses other [heading tags](https://www.markdownguide.org/basic-syntax/#headings) in order (meaning it does not skip numbers). 

## Text

- [ ] All link text is descriptive. It tells users where they will be taken if they open the link.
- [ ] All acronyms are defined at least the first time they are used. 
- [ ] Field-specific/specialized terms are used when needed, but not excessively.

## Code

- [ ] Code sections are introduced and explained before they appear in the notebook. This can be fulfilled with a heading in a prior Markdown cell, a sentence preceding it, or a code comment in the code section.
- [ ] Code has explanatory comments (if relevant). This is most important for long sections of code.
- [ ] If the author has control over the syntax highlighting theme in the notebook, that theme has enough color contrast to be legible.
- [ ] Code and code explanations focus on one task at a time. Unless comparison is the point of the notebook, only one method for completing the task is described at a time.

### Images

- [ ] All images (jpg, png, svgs) have an [image description](https://www.w3.org/WAI/tutorials/images/decision-tree/). This could be
    - [ ] [Alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt) (an `alt` property) 
    - [ ] [Empty alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt#decorative_images) for decorative images/images meant to be skipped (an `alt` attribute with no value)
    - [ ] Captions
    - [ ] If no other options will work, the image is decribed in surrounding paragraphs.

- [ ] Any [text present in images](https://www.w3.org/WAI/WCAG21/Understanding/images-of-text.html) exists in a text form outside of the image (this can be alt text, captions, or surrounding text.)

### Visualizations

- [ ] All visualizations have an image description. Review the previous section, Images, for more information on how to add it.
- [ ] [Visualization descriptions](http://diagramcenter.org/specific-guidelines-e.html) include
    - [ ] The type of visualization (like bar chart, scatter plot, etc.)
    - [ ] Title
    - [ ] Axis labels and range
    - [ ] Key or legend
    - [ ] An explanation of the visualization's significance to the notebook (like the trend, an outlier in the data, what the author learned from it, etc.)

- [ ] All visualizations and their parts have [enough color contrast](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) ([color contrast checker](https://webaim.org/resources/contrastchecker/)) to be legible. Remember that transparent colors have lower contrast than their opaque versions.
- [ ] All visualizations [convey information with more visual cues than color coding](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html). Use text labels, patterns, or icons alongside color to achieve this.
- [ ] All visualizations have an additional way for notebook readers to access the information. Linking to the original data, including a table of the data in the same notebook, or sonifying the plot are all options.

### CODE REVIEW

**Comment by drlaw1558** [comment]
Tagging @rizeladiaz as well; I'd like to merge this update asap as it looks like the latest pipeline has a bug such that it does not work with the previous notebook (bug in multiprocessing) and we're starting to get multiple helpdesk calls about it.

**Comment by sosey** [comment]
@drlaw1558. should there be a notebook in the caveate repo for the bug workaround? And does this updated notebook work around the bug or change to an example that doesn't exercise it?

**Comment by drlaw1558** [comment]
Bug workaround is trivial and doesn't need a notebook, the answer is to just turn off multiprocessing.  This is off by default in the pipeline anyway, it's just the old notebook that had it enabled (which I've already turned off in the new notebook as it's unnecessary any longer).

File: notebooks/MIRI/JWPipeNB-MIRI-MRS.ipynb [hunk 1]
```diff
@@ -23,8 +23,8 @@
    "metadata": {},
```
Comment (sosey) [inline notebooks/MIRI/JWPipeNB-MIRI-MRS.ipynb:23]: <pre class="ql-syntax" spellcheck="false">Line #26.    # Turn on multi-core processing for jump step (off by default).  Choose what fraction of cores to use (quarter, half, or all)
</pre><p>Do you still mean this to say turn on multi-core processing? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/jwst-pipeline-notebooks/pull/20/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (sosey) [inline notebooks/MIRI/JWPipeNB-MIRI-MRS.ipynb:23]: <pre class="ql-syntax" spellcheck="false">Line #24.        nselfcal = len(selfcalfiles)
</pre><p>selfcalfiles is a required param and you're expecting a list, yes?  Why not something like </p><p> </p><p>for name in selfcalfiles:</p><p>	 with fits.open(name) as hdu:</p><p>          hdu.verify()</p><p>          if (hdu[0].header['CHANNEL'] == this_channel):</p><p>              asn['products'][0]['members'].append({'expname': name, 'exptype': 'selfcal'})</p><p>if selfcalfiles:</p><p>   write the asn</p><p> </p><p>same comment for the bgfiles lines above this</p><p>           </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/jwst-pipeline-notebooks/pull/20/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (sosey) [reply notebooks/MIRI/JWPipeNB-MIRI-MRS.ipynb:23]: this formatted poorly, but you get the idea :)

**Review by sosey — CHANGES_REQUESTED** [review]
see suggested code change

File: notebooks/MIRI/JWPipeNB-MIRI-MRS.ipynb (hunk 1 above)
Comment (drlaw1558) [reply notebooks/MIRI/JWPipeNB-MIRI-MRS.ipynb:23]: Yes, it's fine for the jump step, just not for ramp fitting.

Comment (drlaw1558) [reply notebooks/MIRI/JWPipeNB-MIRI-MRS.ipynb:23]: Looks like that should be functionally equivalent, just cuts down on some unnecessary variables and vertical space.  Done.

**Comment by rizeladiaz** [comment]
I have been working on this notebook since yesterday but it is failing the pytest. Was thinking about running it in Linux but i can see more changes, correct?

**Comment by drlaw1558** [comment]
Changes should be pretty minimal- options added for a new step, some changes to defaults, and some syntax changes .  I'd anticipate that the pytest will keep failing just like the prior version until we get the runners upgraded.

**Comment by drlaw1558** [comment]
Pinging @sosey @rizeladiaz again; we're continuing to get tickets from users about this.

File: notebooks/MIRI/JWPipeNB-MIRI-MRS.ipynb (hunk 1 above)
Comment (sosey) [reply notebooks/MIRI/JWPipeNB-MIRI-MRS.ipynb:23]: yup, it's more pythonic too

**Comment by sosey** [comment]
@drlaw1558 will you rebase your branch?

**Comment by drlaw1558** [comment]
Ok, rebased

**Review by sosey — APPROVED** [review]
I'm ok with the current version. Next time we update let's change the remaining variable loops to the pythonic versions, that doesn't need to be done this time though.

**Comment by drlaw1558** [comment]
Added some revisions based on comments from @rizeladiaz .  Looks like all approvals now in place.

**Pull request merged**